src/plugin.scss

```diff
@@ -19,182 +34,206 @@ $main-color: #fff;
+    text-shadow:  1px  1px 2px black,
+                 -1px  1px 2px black,
+                  1px -1px 2px black,
+                 -1px -1px 2px black;
   }
 
   .vjs-playlist-name {
     display: block;
-    font-style: normal;
     max-height: 2.5em;
-    overflow: hidden;
-
-    // line-height:normal was causing overflow cutoff issues on Android, since normal
-    // lets different user agents pick a value. Here we set a consistent precise value.
-    line-height: 20px;
 
     // So drop shadow doesn't get cut off as overflow
     padding: 0 0 4px 2px;
+    font-style: normal;
+    text-overflow: ellipsis;
+
+    // line-height: normal was causing overflow cutoff issues on Android, since normal
+    // lets different user agents pick a value. Here we set a consistent precise value.
+    line-height: 20px;
   }
 
   .vjs-playlist-description {
+    margin: 0;
     text-overflow: ellipsis;
     overflow: hidden;
-    margin: 0;
   }
-}
 
-/* Prevent interaction with the playlist menu while ads are playing */
-/* on browsers that don't support pointer-events (IE<11), prevent
-scrolling past the ad overlay */
-.vjs-playlist.vjs-ad-playing {
-  overflow: hidden;
-}
-/* prevent clicks and scrolling from affecting the playlist during ads */
-.vjs-playlist.vjs-ad-playing.vjs-csspointerevents {
-  pointer-events: none;
-  overflow: auto;
+  .vjs-up-next-text {
+    display: none;
+    padding: .1rem 2px;
+    font-size: .8em;
+    text-transform: uppercase;
+  }
 
-  // Fix for IE11 pointer-events bug that allowed clicking playlist during ads
-  .vjs-playlist-ad-overlay {
-    pointer-events: auto;
+  .vjs-up-next {
+
+    .vjs-up-next-text {
+      display: block;
+    }
   }
-}
-/* darken the playlist menu display to indicate it's not interactive
-during ads and capture click events on IE<11 */
-.vjs-playlist.vjs-ad-playing .vjs-playlist-ad-overlay {
-  display: none;
-}
-.vjs-playlist.vjs-ad-playing .vjs-playlist-ad-overlay {
-  display: block;
-  position: absolute;
-  top: 0;
-  left: 0;
-  width: 100%;
-  height: 100%;
 
-  /* IE8 fallback */
-  background-color: #1a1a1a;
-  -ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=50)";
+  // Selected item rules
+  .vjs-selected {
+    background-color: $highlight-color;
 
-  /* modern browsers */
-  background-color: rgba(0, 0, 0, 0.5);
+    img {
+      opacity: .2;
+    }
+
+    .vjs-playlist-duration {
+      display: none;
+    }
+
+    .vjs-playlist-now-playing-text {
+      display: block;
+    }
+
+    .vjs-playlist-title-container {
+      text-shadow: none;
+    }
+  }
 }
 
-/* Parametric rules. These are specialized for touch and mouse-based devices */
-@mixin playlist-base($base-font-size: 14px) {
-  $small-font-size: ceil($base-font-size * 0.75);
-  $item-height: $base-font-size * 4;
-  $name-height: $base-font-size;
-  $description-height: $item-height - $name-height;
-  $thumbnail-font-size: 18px;
+// Vertical/default playlist orientation
+.vjs-playlist-vertical {
```

Comment:
Another key change is splitting out orientation styles based on the configured orientation of the playlist. This means less overriding and clearer CSS.